src/daft-local-execution/src/pipeline.rs

```diff
@@ -817,83 +818,93 @@ fn physical_plan_to_pipeline(
             // 2. Join type. Different join types have different requirements for which side can build the probe table.
             let left_stats_state = left.get_stats_state();
             let right_stats_state = right.get_stats_state();
-            let build_on_left = match join_type {
-                // Inner and outer joins can build on either side. If stats are available, choose the smaller side.
-                // Else, default to building on the left.
-                JoinType::Inner | JoinType::Outer => match (left_stats_state, right_stats_state) {
-                    (
-                        StatsState::Materialized(left_stats),
-                        StatsState::Materialized(right_stats),
-                    ) => {
-                        let left_size = left_stats.approx_stats.num_rows;
-                        let right_size = right_stats.approx_stats.num_rows;
-                        left_size <= right_size
+            let build_on_left = match build_on_left {
```

Comment:
1. could we update the above doc as well? And regarding the description "Probe tables should be built on the smaller side.", should be "built on the bigger side"?, I mean build table is the smaller table, probe table is the bigger table?
2. And if we support passing `build_on_left` directly, could we move the below logic to `hash_join.rs`, otherwise, it's a little bit couple that only the boradcast_join pass this value, but hash_join not. Or could we  consider add a new `LocalPhysicalPlan::Broadcast` refer to the `JoinStrategy::Broadcast`, since we have LocalPhysicalPlan:HashJoin refer to JoinStrategy::Hash & JoinStrategy::Broadcast, and LocalPhysicalPlan:CrossJoin refer to JoinStrategy::Cross?

Reply:
ok，I think it's fine. Regarding 2, I thought move the logical about "calculate build_on_left" to daft-distributed/src/pipeline_node/hash_join.rs, and I realized this only works for distributed mode later, daft-local-plan/src/translate.rs also need do same thing, so no need to move out.